js/components/portal-dashboard/questions/multiple-choice-question.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { PureComponent } from "react";
+import { Map } from "immutable";
+import { renderHTML } from "../../../util/render-html";
+
+import css from "../../../../css/portal-dashboard/questions/multiple-choice-question.less";
+
+interface IProps {
+  question?: Map<string, any>;
+}
+
+export const MultipleChoiceQuestion: React.FC<IProps> = (props) => {
+    const { question } = props;
+    const prompt = question?.get("prompt");
+    const choices: Map<string,any> = question?.get("choices") || [];
+    const renderMultipleChoiceChoices = (choices: Map<string, any>, i: number) => {
+      const multipleChoiceContent = `${choices.get("content")} ${choices.get("correct") ? "(correct)" : ""}`;
+      const multipleChoiceContentClass = `${css.mcContent} ${choices.get("correct") ? css.correct : ""}`;
+
+      return (
+        <div className={css.choiceWrapper} key={`choices ${i}`}>
+          <div className={css.choiceIcon} />
+          <div className={multipleChoiceContentClass}>
+            {multipleChoiceContent}
+          </div>
+        </div>
+      );
+    };
+
+    return (
+      <div className={css.questionText}>
+        {prompt && renderHTML(prompt)}
+        { choices.toArray?.().map(renderMultipleChoiceChoices) }
+      </div>
+    );
```

Comment:
Might be clearer to try to collapse this so all the render is in one return.  I know that's something that @kswenson mentioned a couple times about many of our React class components - that we have sprawling render functions that are hard to follow.

Reply:
Here, this should do the trick:
```
    return (
      <div className={css.questionText}>
        { prompt && renderHTML(prompt) }
        { choices.toArray?.().map((choices: Map<string, any>, i: number) => {
            const multipleChoiceContent = `${choices.get("content")} ${choices.get("correct") ? "(correct)" : ""}`;
            const multipleChoiceContentClass = `${css.mcContent} ${choices.get("correct") ? css.correct : ""}`;
            return (
              <div className={css.choiceWrapper} key={`choices ${i}`}>
                <div className={css.choiceIcon} />
                <div className={multipleChoiceContentClass}>
                  {multipleChoiceContent}
                </div>
              </div>
            );
          })
        }
      </div>
    );
```

Reply:
Replace your `return` in the functional component with the above.  Then you can remove the `renderMultipleChoiceChoices`.  It will be easier to read and might make code climate happy!